From 48abffc7beea8884717bcb631f921a1b7f58bb6d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mat=C4=9Bj=20Cepl?= <mcepl@suse.com>
Date: Mon, 25 Mar 2024 20:15:19 +0100
Subject: [PATCH 1/2] Make use of devpi_process optional

Also skip tests using the enable_pip_pypi_access_fixture, as these tests require
the devpi server and can break without it
---
 src/tox/pytest.py       | 6 +++++-
 tests/test_provision.py | 5 ++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

Index: tox-4.34.1/src/tox/pytest.py
===================================================================
--- tox-4.34.1.orig/src/tox/pytest.py
+++ tox-4.34.1/src/tox/pytest.py
@@ -17,7 +17,10 @@ from typing import TYPE_CHECKING, Any, P
 
 import pytest
 from _pytest.fixtures import SubRequest  # noqa: PLC2701
-from devpi_process import IndexServer
+try:
+    from devpi_process import IndexServer
+except ImportError:
+    IndexServer = None
 from virtualenv.info import fs_supports_symlink
 
 import tox.run
@@ -514,6 +517,7 @@ def enable_pip_pypi_access_fixture(
     """Set a fake pip index url, tests that want to use a pypi server should create and overwrite this."""
     _, previous_url = disable_pip_pypi_access
     enable_pypi_server(monkeypatch, previous_url)
+    pytest.skip()
     return previous_url
 
 
Index: tox-4.34.1/tests/test_provision.py
===================================================================
--- tox-4.34.1.orig/tests/test_provision.py
+++ tox-4.34.1/tests/test_provision.py
@@ -19,7 +19,10 @@ if TYPE_CHECKING:
     from collections.abc import Callable, Iterator, Sequence
 
     from build import DistributionType
-    from devpi_process import Index, IndexServer
+    try:
+        from devpi_process import Index, IndexServer
+    except ImportError:
+        Index, IndexServer = None, None
 
     from tox.pytest import MonkeyPatch, TempPathFactory, ToxProjectCreator
 
